From 1199c18251b039c80d66b0036e4f22bb2ce7dbd8 Mon Sep 17 00:00:00 2001
From: Bandan Das <bsd@redhat.com>
Date: Wed, 6 Nov 2013 02:11:02 +0100
Subject: [PATCH 55/81] kvmvapic: Enter inactive state on hardware reset

RH-Author: Bandan Das <bsd@redhat.com>
Message-id: <1383703863-4840-2-git-send-email-bsd@redhat.com>
Patchwork-id: 55472
O-Subject: [PATCH RHEL-7.0 qemu-kvm 2/3] kvmvapic: Enter inactive state on hardware reset
Bugzilla: 920021
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

ROM layout may change after reset of devices are hotplugged, so we have
to pick up the physical address again when the ROM is initialized. This
is best achieved by resetting the state to INACTIVE.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit c056bc3f3464cfae1c94b7dd633d3ec13b13b655)
---
 hw/i386/kvmvapic.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/kvmvapic.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/hw/i386/kvmvapic.c b/hw/i386/kvmvapic.c
index a2c7e8f..a96ea85 100644
--- a/hw/i386/kvmvapic.c
+++ b/hw/i386/kvmvapic.c
@@ -504,9 +504,7 @@ static void vapic_reset(DeviceState *dev)
 {
     VAPICROMState *s = DO_UPCAST(VAPICROMState, busdev.qdev, dev);
 
-    if (s->state == VAPIC_ACTIVE) {
-        s->state = VAPIC_STANDBY;
-    }
+    s->state = VAPIC_INACTIVE;
     vapic_enable_tpr_reporting(false);
 }
 
-- 
1.7.1

